examples/render/src/main.rs

```diff
@@ -67,17 +68,147 @@ struct Args {
     #[arg(long = "height", default_value_t = 512)]
     height: u32,
 
-    /// Map mode (e.g. 'static', 'tile', 'continuous')
-    #[arg(short = 'm', long = "mode")]
-    mode: Option<String>,
+    /// Map mode
+    #[arg(short = 'm', long = "mode", default_value = "static")]
+    mode: Mode,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Default, clap::ValueEnum)]
+enum Mode {
+    #[default]
+    /// Once-off still image of an arbitrary viewport
+    Static,
+    /// Once-off still image of a single tile
+    Tile,
+    /// Continually updating map
+    Continuous,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, clap::ValueEnum)]
```

Comment:
this enum can be moved to the main lib, with an optional `Clap` dependency (controlled with a feature like `clap` or `cli`) - i.e. the `clap::ValueEnum` will only be enabled if the feature is set, and you don't need to copy one enum to another.

Reply:
In this case, should I revert the decision on using workspace crates instead of cargo examples as well?
https://github.com/nyurik/maplibre-native-rs/pull/11#issuecomment-2727618682

Reply:
Scratch that. This would require support for this in cxx
![image](https://github.com/user-attachments/assets/3a5ec263-5a65-4703-9b05-29a1eee4af98)

Some guy named yuri even commented on this exact issue ^^

- https://github.com/dtolnay/cxx/issues/1022

Reply:
Lol!